frontend/public/components/terminal.jsx

```diff
@@ -33,6 +35,45 @@ export class Terminal extends React.Component {
     this.terminal && this.terminal.focus();
   }
 
+  enterFullScreen() {
+    const masthead = document.querySelector('[role="banner"]');
+    const sidebar = document.getElementById('sidebar');
+    masthead.style.visibility = sidebar.style.visibility = 'hidden';
+
+    this.terminal.toggleFullScreen(true);
+    this.reset();
+    this.terminal.write('Press [Esc] to exit full screen\r\n\nsh-4.2$ ');
+    this.onResize();
+    this.focus();
+  }
+
+  exitFullScreen() {
+    const masthead = document.querySelector('[role="banner"]');
+    const sidebar = document.getElementById('sidebar');
+    masthead.style.visibility = sidebar.style.visibility = 'visible';
+
+    this.terminal.off('key');
+    this.terminal.toggleFullScreen(false);
+    this.reset();
+    this.terminal.write('sh-4.2$ ');
+    this.onResize();
+    this.focus();
+  }
+
+  launchFullScreen () {
+    this.terminal.on('key', (key) => {
+      // if esc key
+      // eslint-disable-next-line no-console
+      console.log(`key = ${ JSON.stringify(key, null, ' ')}`);
+      // eslint-disable-next-line no-console
+      console.log(`key = ${ key.charCodeAt()}`);
+      if (key.charCodeAt() === 27) {
```

Comment:
`if(isEscapeKey(key)) {` might be nice.  So later readers don't have to look up the key code.